contracts/src/internal/HyperdriveCheckpoint.sol

```diff
@@ -238,28 +248,32 @@ abstract contract HyperdriveCheckpoint is
         (uint256 lpSharePrice, ) = _calculateLPSharePriceSafe(_vaultSharePrice);
         emit CreateCheckpoint(
             _checkpointTime,
+            checkpointVaultSharePrice,
             _vaultSharePrice,
             maturedShortsAmount,
             maturedLongsAmount,
             lpSharePrice
         );
 
-        return _vaultSharePrice;
+        return checkpointVaultSharePrice;
     }
 
     /// @dev Calculates the proceeds of the holders of a given position at
     ///      maturity.
     /// @param _bondAmount The bond amount of the position.
     /// @param _vaultSharePrice The current vault share price.
-    /// @param _openVaultSharePrice The vault share price at the beginning of
-    ///        the position's checkpoint.
+    /// @param _openVaultSharePrice The vault share price from the position's
+    ///        starting checkpoint.
+    /// @param _closeVaultSharePrice The vault share price from the position's
+    ///        ending checkpoint.
     /// @param _isLong A flag indicating whether or not the position is a long.
     /// @return shareProceeds The proceeds of the holders in shares.
     /// @return governanceFee The fee paid to governance in shares.
     function _calculateMaturedProceeds(
         uint256 _bondAmount,
-        uint256 _vaultSharePrice,
         uint256 _openVaultSharePrice,
+        uint256 _closeVaultSharePrice,
+        uint256 _vaultSharePrice,
         bool _isLong
     ) internal view returns (uint256 shareProceeds, uint256 governanceFee) {
```

Comment:
Why does the [shareProceeds](https://github.com/delvtech/hyperdrive/pull/924/files#diff-1f921bd076bd7a60928e5d64b14a74f3f8245316baf175a7f7cfe12ae3d846c8R285) calculation uses `_vaultSharePrice` instead of `_closeVaultSharePrice`?

Logically I would expect proceeds to be calculated based on the price at the maturation checkpoint, rather than the current price, for a mature position.